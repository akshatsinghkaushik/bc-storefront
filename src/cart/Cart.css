.Cart {
  height: max-content;
  min-height: calc(100vh - 4.1rem);
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background-color: rgba(223, 223, 223, 0.795);
}

.cart-title {
  padding: 4rem 0 4rem 0;
}

.product-list {
  margin: 5rem 7rem 5rem 7rem;
}

.cart-items {
  padding: 0 7% 0 7%;
  background-color: white;
}

.product-item {
  padding: 2rem 0 0 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  color: rgb(131, 129, 129);
}

.amount {
  display: flex;
  height: 3rem;
}

.cart-item-text h3,
p {
  margin: 0;
}

.cart-details {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.cart-overview {
  padding: 2rem 0 2rem 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.cart-overview-container {
  transform: translateX(-3rem);
}
.cart-overview-container p {
  margin: 0 0 1rem 0;
  color: rgb(160, 158, 158);
}

.cart-overview-details {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.cart-overview-details .titles {
  margin: 0 12rem 0 0;
}

.cart-overview-details .prices {
  text-align: end;
  transform: translateX(1rem);
}

.quantity,
.total,
.action {
  margin: 0 1rem 0 1rem;
  display: flex;
  align-items: center;
}

.cart-total {
  display: flex;
  align-items: center;
  justify-content: right;
}

.cart-total p {
  display: flex !important;
  justify-self: right !important;
}
