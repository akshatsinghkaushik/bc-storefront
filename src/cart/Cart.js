import React, { useContext } from 'react';
import './Cart.css';
import CartContext from '../context/CartContext';
import CartFooter from './CartFooter';
import CartItems from './CartItems';

const Cart = ({}) => {
  const cartContext = useContext(CartContext);

  return (
    <div className="Cart">
      <header className="cart-title">
        <h1>Shopping Cart</h1>
      </header>
      <div className="cart-items">
        <div className="product-item">
          <p className="">Product</p>
          <div className="cart-details">
            <p className="quantity">Quantity</p>
            <p className="total">Total</p>
            <p className="action">Action</p>
          </div>
        </div>
        <CartItems />
        <hr style={{ height: '1px', width: '100%', margin: '0' }} />
        <div className="cart-overview">
          <div></div>
          <div className="cart-overview-container">
            <p>CART OVERVIEW</p>
            <div className="cart-overview-details">
              <div className="titles">
                <p>SUBTOTAL</p>
                <br />
                <p>TOTAL</p>
              </div>
              <div className="prices">
                <p>$ {Number(cartContext.getCartTotal()).toFixed(2)}</p>
                <br />
                <p style={{ fontSize: '14px', color: 'black' }}>
                  ${Number(cartContext.getCartTotal()).toFixed(2)} CAD
                </p>
              </div>
            </div>
          </div>
        </div>
        <hr style={{ height: '1px', width: '100%', margin: '0' }} />
        <CartFooter />
      </div>
    </div>
  );
};

export default Cart;
