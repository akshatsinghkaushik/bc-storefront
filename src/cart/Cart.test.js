import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Cart from './Cart';
import MockCartProvider from '../tests/MockCartProvider';

it('Cart Renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MockCartProvider>
          <Cart />
        </MockCartProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
