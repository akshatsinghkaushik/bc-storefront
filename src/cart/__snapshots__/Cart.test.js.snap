// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Renders correctly 1`] = `
<div
  className="Cart"
>
  <header
    className="cart-title"
  >
    <h1>
      Shopping Cart
    </h1>
  </header>
  <div
    className="cart-items"
  >
    <div
      className="product-item"
    >
      <p
        className=""
      >
        Product
      </p>
      <div
        className="cart-details"
      >
        <p
          className="quantity"
        >
          Quantity
        </p>
        <p
          className="total"
        >
          Total
        </p>
        <p
          className="action"
        >
          Action
        </p>
      </div>
    </div>
    <div
      className="cart-items-grid"
    >
      <p
        style={
          Object {
            "padding": "2rem",
            "textAlign": "center",
          }
        }
      >
        No Items in Cart
      </p>
    </div>
    <hr
      style={
        Object {
          "height": "1px",
          "margin": "0",
          "width": "100%",
        }
      }
    />
    <div
      className="cart-overview"
    >
      <div />
      <div
        className="cart-overview-container"
      >
        <p>
          CART OVERVIEW
        </p>
        <div
          className="cart-overview-details"
        >
          <div
            className="titles"
          >
            <p>
              SUBTOTAL
            </p>
            <br />
            <p>
              TOTAL
            </p>
          </div>
          <div
            className="prices"
          >
            <p>
              $ 
              0.00
            </p>
            <br />
            <p
              style={
                Object {
                  "color": "black",
                  "fontSize": "14px",
                }
              }
            >
              $
              0.00
               CAD
            </p>
          </div>
        </div>
      </div>
    </div>
    <hr
      style={
        Object {
          "height": "1px",
          "margin": "0",
          "width": "100%",
        }
      }
    />
    <div
      className="cart-footer"
    >
      <a
        className="continue-shopping"
        href="/"
        onClick={[Function]}
      >
        CONTINUE SHOPPING
      </a>
      <button
        className="cart-footer-button"
      >
        CHECKOUT ($
        0.00
        )
      </button>
    </div>
  </div>
</div>
`;
